### DISCUSSION TITLE
Don't set elementary stylesheet system-wide

### DISCUSSION THREAD

**Discussion by cassidyjames** [elementary/default-settings#164] [Ideas]
For elementary OS 6 we've been considering leaving the system stylesheet as Adwaita, and then handling setting the elementary stylesheet in native apps (maybe as a part of Granite?) and in the Flatpak runtime. This way if apps explicitly support elementary OS, they will look correct. But other apps (including those from the repos) will get the "default" GTK styling of Adwaita.

**Comment by worldofpeace** [comment]
This is a great idea :+1:

**Comment by fayjies** [comment]
Probably not a great idea. I hope, the elementary team will reconsider it. As a user of the elementary OS over the last five years, I like its design and consistency; mainly the integration. You all have given GTK apps a new life (even from repos). 
I don't prefer adwaita and I am sure a lot of users too. Also, it will break the design which will make it hard for users to use a system with consistent design. 
Elementary OS is becoming the first choice for Robotics researchers, ML engineers and designers, and all because of its stability, speed but above all, its system-wide consistent design with all GTK-based apps (including programming, technical, design, and research applications). Hope, the elementary team will consider my suggestion and re-consider this decision.

**Comment by worldofpeace** [comment]
That is an interesting take. I'm not sure if they'll go the route where this opt-in is overridable by the user and the distributor. I think for elementary OS it seems like a sensible default.

**Comment by ricksonsa** [comment]
Please do not change this incredible design that elementary os hera have.

**Comment by cassidyjames** [comment]
@ricksonsa we have a new stylesheet in elementary OS 6 that builds on the design of elementary OS while supporting a handful of great new features. This issue is specifically about apps that were not designed for elementary OS, but we're not sure whether or not we're going this route.

For anyone else lurking and wanting more background, a long-standing complaint/issue of elementary OS is that apps designed explicitly for GNOME don't always look right—or in worst cases, are unusable—with the elementary stylesheet. While GNOME and elementary OS are both GTK-based, they have different human interface guidelines, UI patterns, widgets, etc. that make it impossible to guarantee all apps will look as intended in both stylesheets. GNOME apps do seem to work _better_ under the new stylesheet than previous versions, but it's still impossible to guarantee compatibility. As such, one option we have is to let apps that are designed for GNOME use the GNOME stylesheet (Adwaita) instead of trying to force them to use the elementary one. We're not completely decided on this, however.

**Comment by mchalupiak** [comment]
I think the right decision would be to make it opt in for the user. Whether globally or per app somehow, I think this is the best way to cater to the largest group of users. I know for me I would prefer everything to use the eOS style for the consistency, but some might disagree.

**Comment by cassidyjames** [comment]
An update to how I _think_ this is going, based on several online and offline discussions:

1. I think the plan now is to continue to set the stylesheet system-wide by default, especially with support for accent colors baked into elementary OS 6 via the stylesheet itself. This means any natively-running, non-containerized GTK3 apps (like those installed from the Ubuntu repos) will continue to get the elementary stylesheet on elementary OS, as they always have. 

2. However, I believe we plan to _not_ have the elementary stylesheet on Flathub, so cross-platform apps will use Adwaita by default, or can fall back to whatever style they would like. 
  
   1. This is still an open question, however, because GTK does not currently handle falling back to Adwaita well when the app requests a dark stylesheet variant; I consider that a GTK bug, but if it's widespread in use, it will look/feel like an elementary OS bug.

   2. A possible alternative is to let the elementary stylesheet exist on Flathub, but more actively encourage third-party developers of cross-platform Flatpak apps to explicitly test against the elementary stylesheet and/or only allow specific tested stylesheets; the end result would be fewer alien-looking apps, plus some less-consistent but not broken-looking apps.

3. The elementary Flatpak platform will include the elementary Stylesheet, so Flatpak apps explicitly targeting elementary OS and its SDK will use it as expected.

This all may change, but it's how I understand the current situation.

**Comment by JMoerman** [comment]
>    2. A possible alternative is to let the elementary stylesheet exist on Flathub, but more actively encourage third-party developers of cross-platform Flatpak apps to explicitly test against the elementary stylesheet and/or only allow specific tested stylesheets; the end result would be fewer alien-looking apps, plus some less-consistent but not broken-looking apps.

How should I interpret this? Checking `Gtk.Settings.gtk_theme_name` against tested stylesheets and falling back to Adwaita if an untested theme is used?

I would personally prefer the elementary stylesheet to be available on Flathub, the majority of 3rd party apps I have tested here on elementary OS 5 look fine (still look like GNOME apps, though). Requiring developers to explicitly test their apps against the elementary stylesheet sounds reasonable, though. Having said this: I also use some Qt apps and I'm not too bothered by the stylesheet difference with those apps (Qt apps not using CSD helps quite a bit in this regard, though).

**Comment by dominichayesferen** [comment]
> An update to how I _think_ this is going, based on several online and offline discussions:
> 
> 1. I think the plan now is to continue to set the stylesheet system-wide by default, especially with support for accent colors baked into elementary OS 6 via the stylesheet itself. This means any natively-running, non-containerized GTK3 apps (like those installed from the Ubuntu repos) will continue to get the elementary stylesheet on elementary OS, as they always have.
> 2. However, I believe we plan to _not_ have the elementary stylesheet on Flathub, so cross-platform apps will use Adwaita by default, or can fall back to whatever style they would like.
>    
>    1. This is still an open question, however, because GTK does not currently handle falling back to Adwaita well when the app requests a dark stylesheet variant; I consider that a GTK bug, but if it's widespread in use, it will look/feel like an elementary OS bug.
>    2. A possible alternative is to let the elementary stylesheet exist on Flathub, but more actively encourage third-party developers of cross-platform Flatpak apps to explicitly test against the elementary stylesheet and/or only allow specific tested stylesheets; the end result would be fewer alien-looking apps, plus some less-consistent but not broken-looking apps.
> 3. The elementary Flatpak platform will include the elementary Stylesheet, so Flatpak apps explicitly targeting elementary OS and its SDK will use it as expected.
> 
> This all may change, but it's how I understand the current situation.

For 2. ii:
Another route you could go with this idea, if you consider it, is using the Flatpak overrides feature to force the GTK_THEME environment variable in certain non-elementary OS Flatpaks to be 'Adwaita:light' for Adwaita Light or 'Adwaita:dark' for Adwaita Dark.
I've already mentioned how this can be done to Daniel Fore, after discovering it myself, but tl;dr:
1. Use Flatpak's overrides management command to give a specific Flatpak ID a forced environment variable of 'GTK_THEME=Adwaita:light', or similar
2. Run the Flatpak to find it's now locked into using Adwaita only, and will not respect any XSettings changes to the GTK theme for the rest of that Flatpak instance's lifetime.

I'm sure you could implement this into AppCenter and Sideload, somehow, potentially.
Don't forget that overrides can also only be user-wide if you specify the '--user' flag when running flatpak override (in other words for Adwaita Light for that Flatpak user-wide it'd be `flatpak override --user --env=GTK_THEME="Adwaita:light" flatpak.name.here`) so as to not require superuser.

PS: You can do system-wide overrides + user-wide overrides at the same time: the user-wide overrides have priority.

**Comment by dominichayesferen** [comment]
I guess at this point y'all should probably mark a correct answer to close this discussion?
